python-sdk/tests/sql/operators/test_cleanup.py

```diff
@@ -105,12 +105,12 @@ def test_error_raised_with_blocking_op_executors(
     reason="BackfillJobRunner and Job classes are only available in airflow >= 2.6",
 )
 @pytest.mark.parametrize(
-    "executor_in_job,executor_in_cfg,expected_val",
+    "executor_in_job, executor_in_cfg, expected_val",
     [
         (SequentialExecutor(), "LocalExecutor", True),
         (LocalExecutor(), "LocalExecutor", False),
-        (None, "LocalExecutor", False),
-        (None, "SequentialExecutor", True),
+        (LocalExecutor(), "LocalExecutor", False),
```

Comment:
I think line 111 and 112 are duplicate.

Reply:
Do we know which executor was getting used when None was passed?

Reply:
I looked a bit deeper into the issue and discovered that it's not actually the default value causing the problem, but rather caching.

We eventually call https://github.com/apache/airflow/blob/main/airflow/executors/executor_loader.py#L90 when we are doing `job.executor` in `_get_executor_from_job_id` which first time load the value from next call it returns value from https://github.com/apache/airflow/blob/main/airflow/executors/executor_loader.py#L98 in case we have none value for `executor_in_job`. This behavior leads to the last test consistently failing. For instance,
if test sequence is 
```
(None, "LocalExecutor", False),
(None, "SequentialExecutor", True),
```
(None, "SequentialExecutor", True) fail

for sequence
```
(None, "SequentialExecutor", True),
(None, "LocalExecutor", False),
```
(None, "LocalExecutor", False) fail

Reply:
Update the code PTAL commit: e3d3239553a908bca5852525cba8162719a0fc38